### PR TITLE
Add feature blog post for sidecar containers

### PR DESCRIPTION
**Pull request by matthyx**

### CODE REVIEW

File: content/en/blog/_posts/2023-08-15-sidecar-kep.md
Comment (gjkim42) [inline content/en/blog/_posts/2023-08-15-sidecar-kep.md:113]: Can we rebase the branch to add `API for sidecar containers` link that we previously included in https://github.com/kubernetes/website/pull/41802 somewhere?

Comment (matthyx) [reply content/en/blog/_posts/2023-08-15-sidecar-kep.md:113]: rebased

**Comment by reylejano** [comment]
As this is a feature blog for 1.28 and will be published after the 1.28 release day, please target the `main` branch

**Comment by sftim** [comment]
/hold

Release comms will assign a publication date for this; OK to unhold once that date is set and the article is updated.

We should not merge until after the v1.28 release.

**Comment by Rishit-dagli** [comment]
Do we want to try and merge this PR before the code freeze which starts on Monday?

**Comment by bradmccoydev** [comment]
@matthyx Can you please change the publication date for this blog for the 26th of August, thanks

**Comment by matthyx** [comment]
> @matthyx Can you please change the publication date for this blog for the 26th of August, thanks.  cc @krol3 @ramrodo 

Sure, I'm afk until Monday evening. I will update once back.

File: content/en/blog/_posts/2023-08-15-sidecar-kep.md
Comment (krol3) [inline content/en/blog/_posts/2023-08-15-sidecar-kep.md:4]: Hi @matthyx , here Comm shadow 1.28, following the blog date publication, you can check it here: https://github.com/orgs/kubernetes/projects/140/views/6

```suggestion
date: 2023-08-25
```

**Comment by reylejano** [comment]
> Do we want to try and merge this PR before the code freeze which starts on Monday?

@Rishit-dagli , since this is a feature blog, this can merge after the release and target the main branch

File: content/en/blog/_posts/2023-08-15-sidecar-kep.md
Comment (sftim) [reply content/en/blog/_posts/2023-08-15-sidecar-kep.md:4]: You should rename the file to match the new date, too.

File: content/en/blog/_posts/2023-08-15-sidecar-kep.md
Comment (sftim) [inline content/en/blog/_posts/2023-08-15-sidecar-kep.md:29]: Please only use spaces for indentation.

File: content/en/blog/_posts/2023-08-15-sidecar-kep.md
Comment (sftim) [reply content/en/blog/_posts/2023-08-15-sidecar-kep.md:113]:
```suggestion
- Read [API for sidecar containers](/docs/concepts/workloads/pods/init-containers/#api-for-sidecar-containers)
  in the Kubernetes documentation
- Read the [Sidecar KEP](https://github.com/kubernetes/enhancements/blob/master/keps/sig-node/753-sidecar-containers/README.md)
```

File: content/en/blog/_posts/2023-08-15-sidecar-kep.md
Comment (sftim) [inline content/en/blog/_posts/2023-08-15-sidecar-kep.md:83]: (nit) this would work better as a bulleted list.

File: content/en/blog/_posts/2023-08-15-sidecar-kep.md
Comment (sftim) [inline content/en/blog/_posts/2023-08-15-sidecar-kep.md:71]: (nit)
```suggestion
## Known issues
```

File: content/en/blog/_posts/2023-08-15-sidecar-kep.md
Comment (sftim) [inline content/en/blog/_posts/2023-08-15-sidecar-kep.md:42]:
```suggestion
- Init containers have a well-defined startup order regardless of whether you set a `restartPolicy`, so you can ensure that your sidecar starts before any container declarations that come after the sidecar declaration in your manifest.
```

File: content/en/blog/_posts/2023-08-15-sidecar-kep.md
Comment (sftim) [inline content/en/blog/_posts/2023-08-15-sidecar-kep.md:51]: This needs fixing.

**Comment by sftim** [comment]
/retitle Add feature blog post for sidecar containers

File: content/en/blog/_posts/2023-08-15-sidecar-kep.md
Comment (matthyx) [reply content/en/blog/_posts/2023-08-15-sidecar-kep.md:51]: asked @howardjohn

**Comment by sftim** [comment]
/hold cancel

Let's get a tech review now the article is finished (no TODOs etc).

File: content/en/blog/_posts/2023-08-15-sidecar-kep.md
Comment (gjkim42) [reply content/en/blog/_posts/2023-08-15-sidecar-kep.md]: @matthyx 
Could you address this?

**Comment by gjkim42** [comment]
Thanks!

/lgtm

@matthyx @tzneal
/hold 
in case any of you want to update this before merging.

File: content/en/blog/_posts/2023-08-25-sidecar-kep.md
Comment (gjkim42) [inline content/en/blog/_posts/2023-08-25-sidecar-kep.md:4]: we need to fix this.

File: content/en/blog/_posts/2023-08-15-sidecar-kep.md
Comment (gjkim42) [reply content/en/blog/_posts/2023-08-15-sidecar-kep.md]: Do we have to merge this blog on 08-25?

File: content/en/blog/_posts/2023-08-25-sidecar-kep.md
Comment (sftim) [reply content/en/blog/_posts/2023-08-25-sidecar-kep.md:4]: Why is this date wrong?

Comment (gjkim42) [reply content/en/blog/_posts/2023-08-25-sidecar-kep.md:4]: > Why is this date wrong?

Couldn't this blog be released today?

Comment (sftim) [reply content/en/blog/_posts/2023-08-25-sidecar-kep.md:4]: > Couldn't this blog be released today?

Sure, but you'll need to ask release comms for an exception (and then get that exception).

Comment (gjkim42) [reply content/en/blog/_posts/2023-08-25-sidecar-kep.md:4]: > https://github.com/orgs/kubernetes/projects/140/views/6 lists the 25th of August.

Thanks, I didn't realize that we have the scheduled date :)

**Review by tzneal — APPROVED** [review]
/lgtm

Thanks @matthyx !

**Comment by divya-mohan0209** [comment]
Hello folks, is this ready to be merged please?

**Comment by matthyx** [comment]
yes, but published on the 25th please

**Comment by divya-mohan0209** [comment]
Yep, the merge will happen today as soon as I unhold & the publishing will happen per the date field i.e. 25th August.
/unhold

**Pull request merged**
